### PR TITLE
REGR: Check for float in isnaobj_old

### LINKED ISSUE #35493: BUG: use_inf_as_na options raises value error for read_csv
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample to Reproduce Error

```python
import pandas as pd
import numpy as np
pd.set_option('use_inf_as_na', True)
pd.DataFrame({'test_data':[1,3,4,np.nan]}).to_csv('test_data.csv', na_rep='NaN')
pd.read_csv('test_data.csv',sep=',' ,na_values='NaN')
```
Causes `ValueError`:
```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/io/parsers.py", line 686, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/io/parsers.py", line 458, in _read
    data = parser.read(nrows)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/io/parsers.py", line 1201, in read
    df = DataFrame(col_dict, columns=columns, index=index)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/frame.py", line 467, in __init__
    mgr = init_dict(data, index, columns, dtype=dtype)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/internals/construction.py", line 250, in init_dict
    missing = arrays.isna()
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/series.py", line 4795, in isna
    return super().isna()
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/generic.py", line 7109, in isna
    return isna(self).__finalize__(self, method="isna")
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/dtypes/missing.py", line 124, in isna
    return _isna(obj)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/dtypes/missing.py", line 157, in _isna
    return _isna_ndarraylike(obj, inf_as_na=inf_as_na)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/dtypes/missing.py", line 218, in _isna_ndarraylike
    result = _isna_string_dtype(values, dtype, inf_as_na=inf_as_na)
  File "/home/lena/anaconda3/envs/pandas_pip2/lib/python3.8/site-packages/pandas/core/dtypes/missing.py", line 246, in _isna_string_dtype
    vec = libmissing.isnaobj_old(values.ravel())
  File "pandas/_libs/missing.pyx", line 160, in pandas._libs.missing.isnaobj_old
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

#### Problem description
Nans are not recognized as expected anymore if pandas option `use_inf_as_na`is set to True. Occurred first after upgrading to pandas 1.1.0

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : d9fff2792bf16178d4e450fe7384244e50635733
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-42-generic
Version          : #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.0
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2
setuptools       : 49.2.0.post20200712
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.0
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes #35493
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Looks like we lost an isinstance check that caused this regression.

### CODE REVIEW

File: pandas/_libs/missing.pyx [hunk 1]
```diff
@@ -155,7 +155,10 @@ def isnaobj_old(arr: ndarray) -> ndarray:
     result = np.zeros(n, dtype=np.uint8)
     for i in range(n):
         val = arr[i]
-        result[i] = checknull(val) or val == INF or val == NEGINF
+        result[i] = (
+            checknull(val)
+            or isinstance(val, float) and (val == INF or val == NEGINF)
```
Comment (jbrockmendel) [inline pandas/_libs/missing.pyx:160]: Can you use is_float_object here

**Review by simonjayhawkins — APPROVED** [review]
Thanks @dsaxton lgtm

File: doc/source/whatsnew/v1.1.1.rst [hunk 2]
```diff
@@ -15,7 +15,7 @@ including other versions of pandas.
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
 
--
+- Fixed regression where :func:`read_csv` would raise a ``ValueError`` when ``use_inf_as_na`` was set to ``True`` (:issue:`35493`).
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.1.rst:18]: write out the path to the option

File: pandas/tests/io/parser/test_common.py [hunk 3]
```diff
@@ -2179,3 +2179,13 @@ def test_read_csv_names_not_accepting_sets(all_parsers):
     parser = all_parsers
     with pytest.raises(ValueError, match="Names should be an ordered collection."):
         parser.read_csv(StringIO(data), names=set("QAZ"))
+
+
+def test_read_csv_with_use_inf_as_na(all_parsers):
+    # https://github.com/pandas-dev/pandas/issues/35493
```
Comment (jreback) [inline pandas/tests/io/parser/test_common.py:2185]: see if u can locate this test near others that set options (not kwargs)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/parser/test_common.py (hunk 3 above)
Comment (dsaxton) [reply pandas/tests/io/parser/test_common.py:2185]: The others that use options seem to be in more targeted testing files (test_to_html.py, test_printing.py); this is the only one in this file

**Review by gfyoung — APPROVED** [review]

File: pandas/tests/io/parser/test_common.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/io/parser/test_common.py:2185]: kk

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton

**Comment by jreback** [comment]
backport to 1.1.x

**Comment by jreback** [comment]
@simonjayhawkins hmm these don't seem to be triggering a backport action. any idea?

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins hmm these don't seem to be triggering a backport action. any idea?

it worked on #35452 but not from #35272

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.1.x

**Pull request merged**